### PR TITLE
#4800 Add field for meta author and use it in frontend

### PR DESCRIPTION
**Pull request by Schrank**
### 1. Why is this change necessary?

See #4800 

### 2. What does this change do, exactly?

Add a configuration field to set a meta author and use it in the frontend.

### 3. Describe each step to reproduce the issue or behaviour.

See #4800 

### 4. Please link to the relevant issues (if any).

#4800 

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [ ] ~I have written tests and verified that they fail without my change~
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] ~I have written or adjusted the documentation according to my changes~
- [ ] ~This change has comments for package types, values, functions, and non-obvious lines of code~
- [X] I have read the contribution requirements and fulfill them.

### CODE REVIEW

**Comment by Schrank** [comment]
@shyim

**Comment by seggewiss** [comment]
@shyim for me this looks okay. Should I import this? 🤔

File: src/Storefront/Page/GenericPageLoader.php [hunk 1]
```diff
@@ -91,4 +93,9 @@ public function load(Request $request, SalesChannelContext $context): Page
             return $page;
         });
     }
+
+    public function getSystemConfig(string $config, string $salesChannelId): string
```
Comment (h1k3r) [inline src/Storefront/Page/GenericPageLoader.php]: I would make it private, so it does not become a part of a public interface.

**Comment by shyim** [comment]
@seggewiss yes

File: src/Storefront/Page/GenericPageLoader.php [hunk 2]
```diff
@@ -94,7 +94,7 @@ public function load(Request $request, SalesChannelContext $context): Page
         });
     }
 
-    public function getSystemConfig(string $config, string $salesChannelId): string
+    private function getSystemConfig(string $config, string $salesChannelId): string
```
Comment (Schrank) [inline src/Storefront/Page/GenericPageLoader.php:97]: Thanks! SO YES YES!

**Comment by Schrank** [comment]
I assume this is done - if you need anything else, please tell me :-)

File: src/Storefront/Page/GenericPageLoader.php (hunk 2 above)
Comment (aragon999) [reply src/Storefront/Page/GenericPageLoader.php:97]: Why in general add a new method, in my personal opinion this is an not needed abstraction :-)

(You can still move the `$salesChannelId` into a own variable, which is fine I think).

**Comment by BrocksiNet** [comment]
Hello together :wave: I have a question: why do we need this?

[Here](https://github.com/shopware/shopware/blob/trunk/src/Storefront/Resources/views/storefront/layout/meta.html.twig#L21), we already have some meta tag for the author.

So, in which case do we need that new author field? Is the current one not usable?

Thx for some feedback. :hugs:

**Comment by shyim** [comment]
@BrocksiNet thats always null right now, thats why he added it

File: src/Storefront/Page/GenericPageLoader.php [hunk 3]
```diff
@@ -38,11 +38,13 @@ public function load(Request $request, SalesChannelContext $context): Page
         return Profiler::trace('generic-page-loader', function () use ($request, $context) {
             $page = new Page();
 
+            $salesChannelId = $context->getSalesChannel()->getId();
```
Comment (aragon999) [inline src/Storefront/Page/GenericPageLoader.php:41]:
```suggestion
            $salesChannelId = $context->getSalesChannelId();
```

**Comment by BrocksiNet** [comment]
Okay, but should we not add it here:
image
Maybe the general setting is fallback but with the possibility to overwrite via pages? :thinking: 

Because different pages could also have various authors?

**Comment by mitelg** [comment]
thanks for your contribution @Schrank 👍🎉💙

**Pull request closed without merging**